crypto/src/main/java/org/web3j/crypto/StructuredDataEncoder.java

```diff
@@ -97,9 +106,106 @@ public static String encodeType(String primaryType) {
         return Numeric.hexStringToByteArray(sha3String(encodeType(primaryType)));
     }
 
+    public static List<Integer> getArrayDimensionsFromDeclaration(String declaration) {
+        String arrayTypeRegex = "^[a-zA-Z_$][a-zA-Z_$0-9]*((\\[([1-9]\\d*)?\\])+)$";
```

Comment:
If you use the same regexp constant in encodeData maybe it makes sense to extract it to a class constant.